tests/tree.rs

```diff
@@ -128,6 +130,32 @@ fn single_named_parameter() {
     }
 }
 
+#[test]
+fn repeated_single_named_param() {
+    //  Pattern: /users/:id
+    //
+    //      /users/gordon              match
+    //      /users/you                 match
+    //      /users/gordon/profile      no match
+    //      /users/                    no match
+    let mut tree = PathTree::new();
+
+    tree.insert("/users/:id", 0);
+    tree.insert("/users/:user_id", 1);
+
+    let r = tree.find("/users/gordon");
+    let path = r.unwrap();
+    match (path.value, path.params()) {
+        (0, params) => {
+            assert_eq!(params, vec![("id", "gordon")])
```

Comment:
After being overwritten, this is no longer valid

Reply:
This test is not trying to establish behavior with respect to overriding, what it is saying is that _if_ the matched value is `0` then the param name should be `id` and if the value is `1` the param name should be `user_id`. It is currently failing because the returned value is `1` but the param name is `id` which is wrong. In other words, the only reason I included this branch is to avoid specifying the behavior of overriding. If you already decided that the last parameter is preserved then the test can be simplified to:


```rust
fn repeated_single_named_param() {
    //  Pattern: /users/:id
    //
    //      /users/gordon              match
    //      /users/you                 match
    //      /users/gordon/profile      no match
    //      /users/                    no match
    let mut tree = PathTree::new();

    tree.insert("/users/:id", 0);
    tree.insert("/users/:user_id", 1);

    let r = tree.find("/users/gordon");
    let path = r.unwrap();
   assert_eq!(path.value, 1);
    assert_eq!(path.params, vec![("user_id", "gordon")]);
}
```

Reply:
Yes, this can only be reserved for one case